### ISSUE TITLE
Attribute table filter between dates no result

### ISSUE THREAD

**Issue by dingausmwald** [qgis/QGIS#43177]
Windows 10
QGIS 3.16.6

cant filter between dates in attribute table. My table consists of features with dates formated with german date format (dd.MM.yyyy). When trying to filter in the attribute table with "between (inclusive)" setting in the filter menue, the results are 0, but there are definitly dates between.

**Comment by dingausmwald** [comment]
Sorry this took me so long. Stumbled over this error again. Here is some sampledata. Also happens with "less than" oder "greater than" filter. Works when date format is in default. So i believe this has to be something with the custom format
[QGIS test date error sampledata.zip](https://github.com/qgis/QGIS/files/7235192/QGIS.test.date.error.sampledata.zip)
.

**Comment by dingausmwald** [comment]
Tested with 3.16.11 and 3.20.2

**Comment by signedav** [comment]
Ah, I think I found it. And it's already fixed here #51471

**Comment by tudorbarascu** [comment]
Same here. Tested today with 3.28.4 and the problem resides.
I've attached a test project. [bug.zip](https://github.com/qgis/QGIS/files/11072604/bug.zip)

I configured the widget on the field like this:
![image](https://user-images.githubusercontent.com/3179646/227795763-1c754abc-b27d-4948-a653-83cc5ca5bee5.png)

I would have expected that entering in the attribute filter the exact custom format would filter to the correct result but I get no filtered feature.
